xds/internal/xdsclient/client.go

```diff
@@ -294,6 +296,25 @@ type HashPolicy struct {
 	RegexSubstitution string
 }
 
+// RouteAction is the action of the route from a received RDS response.
+type RouteAction int
+
+const (
+	// RouteActionRoute is the expected route type on the client side. Route
+	// represents routing a request to some upstream cluster. On the client
+	// side, if an RPC matches to a route that is not RouteActionRoute, the RPC
+	// will fail according to A36.
+	RouteActionRoute RouteAction = iota
+	// RouteActionNonForwardingAction is the expected route type on the server
+	// side. NonForwardingAction represents when a route will generate a
+	// response directly, without forwarding to an upstream host.
+	RouteActionNonForwardingAction
+	// RouteActionUnsupported are routing types currently unsupported by grpc.
+	// According to A36, "A Route with an inappropriate action causes RPCs
+	// matching that route to fail."
+	RouteActionUnsupported
```

Comment:
I originally had this as iota, but switched it back because of test compatibility (i.e. existing tests wouldn't have to change anything). Looking around at client.go, to keep it consistent with the other enums/default value desired, I'll switch it back.

Reply:
Yes, let's do that.

Reply:
![IMG_7931](https://user-images.githubusercontent.com/39203661/128282739-483550c0-f978-44e2-9a02-5d39bb562eee.jpg)
The most recent commit definitely switched these around already :). This comment is on a previous commit.